File: cpp-tiny/lib/Models/Attachments.cpp


#include "Attachments.h"

using namespace Tiny;

Attachments::Attachments()
{
	id = int(0);
	name = std::string();
	path_id = std::string();
	size = int(0);
	create_time = int(0);
	messages = std::list<Attachments_messages>();
}

Attachments::Attachments(std::string jsonString)
{
	this->fromJson(jsonString);
}

Attachments::~Attachments()
{

}

void
Attachments::fromJson(std::string jsonObj)
{   
    bourne::json object = bourne::json::parse(jsonObj);

    const char *idKey = "id";
    
    if(object.has_key(idKey)) 
    {
        bourne::json value = object[idKey];


        
        jsonToValue(&id, value, "int");


    }

    const char *nameKey = "name";
    
    if(object.has_key(nameKey)) 
    {
        bourne::json value = object[nameKey];


        
        jsonToValue(&name, value, "std::string");


    }

    const char *path_idKey = "path_id";
    
    if(object.has_key(path_idKey)) 
    {
        bourne::json value = object[path_idKey];


        
        jsonToValue(&path_id, value, "std::string");


    }

    const char *sizeKey = "size";
    
    if(object.has_key(sizeKey)) 
    {
        bourne::json value = object[sizeKey];


        
        jsonToValue(&size, value, "int");


    }

    const char *create_timeKey = "create_time";
    
    if(object.has_key(create_timeKey)) 
    {
        bourne::json value = object[create_timeKey];


        
        jsonToValue(&create_time, value, "int");


    }

    const char *messagesKey = "messages";
    
    if(object.has_key(messagesKey)) 
    {
        bourne::json value = object[messagesKey];


        std::list<Attachments_messages> messages_list;
        Attachments_messages element;
        for(auto& var : value.array_range())
        {

            
            element.fromJson(var.dump());
            
            messages_list.push_back(element);
        }
        messages = messages_list;


    }


}

bourne::json
Attachments::toJson()
{
    bourne::json object = bourne::json::object();

    



    object["id"] = getId();


    



    object["name"] = getName();


    



    object["path_id"] = getPathId();


    



    object["size"] = getSize();


    



    object["create_time"] = getCreateTime();


    
    

    std::list<Attachments_messages> messages_list = getMessages();
    bourne::json messages_arr = bourne::json::array();

    for(auto& var : messages_list)
    {
        Attachments_messages obj = var;
        messages_arr.append(obj.toJson());
    }
    object["messages"] = messages_arr;




    return object;

}

int
Attachments::getId()
{
	return id;
}

void
Attachments::setId(int  id)
{
	this->id = id;
}

std::string
Attachments::getName()
{
	return name;
}

void
Attachments::setName(std::string  name)
{
	this->name = name;
}

std::string
Attachments::getPathId()
{
	return path_id;
}

void
Attachments::setPathId(std::string  path_id)
{
	this->path_id = path_id;
}

int
Attachments::getSize()
{
	return size;
}

void
Attachments::setSize(int  size)
{
	this->size = size;
}

int
Attachments::getCreateTime()
{
	return create_time;
}

void
Attachments::setCreateTime(int  create_time)
{
	this->create_time = create_time;
}

std::list<Attachments_messages>
Attachments::getMessages()
{
	return messages;
}

void
Attachments::setMessages(std::list <Attachments_messages> messages)
{
	this->messages = messages;
}

File: java-vertx-web/src/main/java/org/openapitools/vertxweb/server/model/Attachments.java
package org.openapitools.vertxweb.server.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.ArrayList;
import java.util.List;
import org.openapitools.vertxweb.server.model.AttachmentsMessages;

/**
 * Dictionary containing details of a file uploaded by a user. 
 **/
@JsonInclude(JsonInclude.Include.NON_NULL)
public class Attachments   {
  
  private Integer id;
  private String name;
  private String pathId;
  private Integer size;
  private Integer createTime;
  private List<AttachmentsMessages> messages = new ArrayList<>();

  public Attachments () {

  }

  public Attachments (Integer id, String name, String pathId, Integer size, Integer createTime, List<AttachmentsMessages> messages) {
    this.id = id;
    this.name = name;
    this.pathId = pathId;
    this.size = size;
    this.createTime = createTime;
    this.messages = messages;
  }

    
  @JsonProperty("id")
  public Integer getId() {
    return id;
  }
  public void setId(Integer id) {
    this.id = id;
  }

    
  @JsonProperty("name")
  public String getName() {
    return name;
  }
  public void setName(String name) {
    this.name = name;
  }

    
  @JsonProperty("path_id")
  public String getPathId() {
    return pathId;
  }
  public void setPathId(String pathId) {
    this.pathId = pathId;
  }

    
  @JsonProperty("size")
  public Integer getSize() {
    return size;
  }
  public void setSize(Integer size) {
    this.size = size;
  }

    
  @JsonProperty("create_time")
  public Integer getCreateTime() {
    return createTime;
  }
  public void setCreateTime(Integer createTime) {
    this.createTime = createTime;
  }

    
  @JsonProperty("messages")
  public List<AttachmentsMessages> getMessages() {
    return messages;
  }
  public void setMessages(List<AttachmentsMessages> messages) {
    this.messages = messages;
  }


  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    Attachments attachments = (Attachments) o;
    return Objects.equals(id, attachments.id) &&
        Objects.equals(name, attachments.name) &&
        Objects.equals(pathId, attachments.pathId) &&
        Objects.equals(size, attachments.size) &&
        Objects.equals(createTime, attachments.createTime) &&
        Objects.equals(messages, attachments.messages);
  }

  @Override
  public int hashCode() {
    return Objects.hash(id, name, pathId, size, createTime, messages);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class Attachments {\n");
    
    sb.append("    id: ").append(toIndentedString(id)).append("\n");
    sb.append("    name: ").append(toIndentedString(name)).append("\n");
    sb.append("    pathId: ").append(toIndentedString(pathId)).append("\n");
    sb.append("    size: ").append(toIndentedString(size)).append("\n");
    sb.append("    createTime: ").append(toIndentedString(createTime)).append("\n");
    sb.append("    messages: ").append(toIndentedString(messages)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}
